langchain/src/vectorstores/closevector_node.ts

```diff
@@ -0,0 +1,286 @@
+  ) {
+    if (!options.credentials.key || !options.credentials.secret) {
+      throw new Error("key and secret must be provided");
+    }
+    const instance = await CloseVectorHNSWNode.loadFromCloud({
+      ...options,
+      accessKey: options.credentials.key,
+      secret: options.credentials.secret,
+    });
+    const vectorstore = new this(options.embeddings, instance.args, options.credentials);
+    return vectorstore;
+  }
+
+  /**
+   * Static method to load a vector store from a directory. It reads the
+   * HNSW index, the arguments, and the document store from the directory,
+   * then creates a new HNSWLib instance with these values.
+   * @param directory The directory from which to load the vector store.
+   * @param embeddings The embeddings to be used by the CloseVectorNode instance.
+   * @returns A Promise that resolves to a new CloseVectorNode instance.
+   */
+  static async load(
+    directory: string,
+    embeddings: Embeddings,
+    credentials?: CloseVectorCredentials
+  ) {
+    const instance = await CloseVectorHNSWNode.load(directory, embeddings);
+    const vectorstore = new this(embeddings, instance.args, credentials);
+    return vectorstore;
+  }
+
+  /**
+   * Method to add vectors to the vector store. It first initializes the
+   * index if it hasn't been initialized yet, then adds the vectors to the
+   * index and the documents to the document store.
+   * @param vectors The vectors to be added to the vector store.
+   * @param documents The documents corresponding to the vectors.
+   * @returns A Promise that resolves when the vectors and documents have been added.
+   */
+  async addVectors(vectors: number[][], documents: Document[]) {
+    await this.instance.addVectors(vectors, documents);
+  }
+
+  /**
+   * Method to perform a similarity search in the vector store using a query
+   * vector. It returns the k most similar documents along with their
+   * similarity scores. An optional filter function can be provided to
+   * filter the documents.
+   * @param query The query vector.
+   * @param k The number of most similar documents to return.
+   * @param filter An optional filter function to filter the documents.
+   * @returns A Promise that resolves to an array of tuples, where each tuple contains a document and its similarity score.
+   */
+  async similaritySearchVectorWithScore(
+    query: number[],
+    k: number,
+    filter?: this["FilterType"]
+  ) {
+    const resp = await this.instance.similaritySearchVectorWithScore(
+      query,
+      k,
+      filter
+        ? (x: CloseVectorDocument<Record<string, unknown>>) =>
+            filter?.({
+              pageContent: x.pageContent,
+              metadata: x.metadata || {},
+            }) || false
+        : undefined
+    );
+    const mapped = resp.map((x) => [
+      {
+        pageContent: x[0].pageContent,
+        metadata: x[0].metadata || {},
+      },
+      x[1],
+    ]) as [Document<Record<string, unknown>>, number][];
```

Comment:
Is cast necessary? Could we just create a `Document` from the `pageContent` and `metadata` and do `as const` instead?

Reply:
Yes, you are correct. I have adjusted it according to your suggestion. Thank you!